### PR TITLE
WIP: Updating scatter lesson

### PR DESCRIPTION
**Pull request by bcosc**
Working on the scatter lesson.

- [x] move 2+ factor scattering to [intermediate](https://github.com/common-workflow-lab/cwl-intermediate-tutorial) repo
- [x] fix formatting

### CODE REVIEW

File: _episodes/scatter.md [hunk 1]
```diff
@@ -1,18 +1,141 @@
...
+    type: File[]
+    outputSource: GATK_HaplotypeCaller/vcf
+steps:
+  GATK_HaplotypeCaller:
+    run: GATK_HaplotypeCaller.cwl
+    in:
+      input_bam: bam
+      intervals: chromosomes
+    out: [vcf]
+~~~
```
Comment (swzCuroverse) [inline _episodes/scatter.md]: So currently this doesn't work, perhaps we should give them a working example with a single chromosome.

File: _episodes/scatter.md [hunk 2]
```diff
@@ -1,18 +1,141 @@
...
+steps:
+  GATK_HaplotypeCaller:
+    run: GATK_HaplotypeCaller.cwl
+    scatter: [intervals]
+~~~
+
+GATK will take the array of chromosome strings, and then using it's intervals input option, create multiple VCFs, one for each chromosome. All these jobs will be run in parallel, and run on as maybe compute nodes available.
+
+When scattering on multiple inputs, you need to explicitly say how the scatter should occur. There are 3 scatter methods in CWL: dot_product, flat_crossproduct and nested_crossproduct. dot_product is the default method, which takes each element of the array and runs on each nth item of the array. flat_crossproduct and nested_crossproduct will take both inputs and run on every combination of both arrays. The difference between flat and nested is in the output type. Flat will create a single array output whereas Nested will create a nested array output.
```
Comment (swzCuroverse) [inline _episodes/scatter.md]: Add images representing flat and nested?

**Comment by mr-c** [comment]
Formatting help https://carpentries.github.io/lesson-example/

**Comment by tobyhodges** [comment]
> Formatting help https://carpentries.github.io/lesson-example/

Also feel free to ping me if you still have questions after reading the above. For more theoretical guidance about designing teaching material, you may find [The Carpentries Curriculum Development Handbook](https://cdh.carpentries.org) helpful. Happy to answer questions about any of this.

File: _episodes/scatter.md [hunk 3]
```diff
@@ -1,18 +1,123 @@
...
+> >     in:
+> >       input_bam: bam
+> >       intervals: chromosomes
+> >     out: [vcf]
+> > GATK will take the array of chromosome strings, and then using it's intervals input option, create multiple VCFs, one for each chromosome. All these jobs will be run in parallel, and run on as maybe compute nodes available.
+> > # TODO: Add diagram of how the job will run
+> {: .solution}
+{: .challenge}
+
+When scattering on multiple inputs, you need to explicitly say how the scatter should occur. There are 3 scatter methods in CWL: dot_product, flat_crossproduct and nested_crossproduct. dot_product is the default method, which takes each element of the array and runs on each nth item of the array. flat_crossproduct and nested_crossproduct will take both inputs and run on every combination of both arrays. The difference between flat and nested is in the output type. Flat will create a single array output whereas Nested will create a nested array output.
```
Comment (mr-c) [inline _episodes/scatter.md]: This is great, lets move scattering of 2+ input to the intermediate repository

File: _episodes/scatter.md [hunk 4]
```diff
@@ -1,18 +1,106 @@
...
+`dot_product`, `flat_crossproduct` and `nested_crossproduct`.
+`dot_product` is the default method,
+which takes each element of the array and runs on each nth item of the array.
+`flat_crossproduct` and `nested_crossproduct` will take both inputs and
+run on every combination of both arrays.
+The difference between flat and nested is in the output type.
+Flat will create a single array output whereas
+nested will create a nested array output.
+
+# TODO Add pictures of cross product / matrix manipulation
```
Comment (mr-c) [inline _episodes/scatter.md]: I think this goes in the intermediate tutorial

File: _episodes/scatter.md [hunk 5]
```diff
@@ -1,18 +1,104 @@
...
+you need to explicitly say how the scatter should occur.
+There are 3 scatter methods in CWL:
+`dot_product`, `flat_crossproduct` and `nested_crossproduct`.
+`dot_product` is the default method,
+which takes each element of the array and runs on each nth item of the array.
+`flat_crossproduct` and `nested_crossproduct` will take both inputs and
+run on every combination of both arrays.
+The difference between flat and nested is in the output type.
+Flat will create a single array output whereas
+nested will create a nested array output.
```
Comment (mr-c) [inline _episodes/scatter.md]:
```suggestion
```

**Review by mr-c — APPROVED** [review]

**Comment by ALuesink** [comment]
PR can be closed: scatter topic is moved to intermediate tutorial

**Pull request merged**